src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,312 @@
+      const backgroundColor =
+        highlightColor !== undefined ? actionBackgroundColor : undefined;
+
+      const styles = useMultiStyleConfig("SubNav", {
+        backgroundColor: backgroundColor,
+        highlightColor: highlightColor,
+      });
+
+      const handleScroll = useCallback(() => {
+        if (scrollableRef.current) {
+          const { scrollLeft, scrollWidth, clientWidth } =
+            scrollableRef.current;
+
+          // Determine if we are scrolling to the left or right
+          const isScrollingRight = scrollLeft > lastScrollLeft;
+          setLastScrollLeft(scrollLeft); // Update last scroll position
+
+          // Show right fade only if scrolling to the left
+          if (!isScrollingRight) {
+            // Show right fade only if not at the end
+            const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+            setShowRightFade(!atRightEdge);
+          } else {
+            setShowRightFade(false); // Hide when scrolling right
+          }
+        }
+      }, [lastScrollLeft, setLastScrollLeft, setShowRightFade]);
+
+      useEffect(() => {
+        const refCurrent = scrollableRef.current;
+        if (refCurrent) {
+          refCurrent.addEventListener("scroll", handleScroll);
+
+          // Initial check to set the right fade effect based on the initial scroll state
+          const { scrollWidth, clientWidth, scrollLeft } = refCurrent;
+          const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+
+          // Set the right fade effect if content is scrollable and not at the right edge
+          setShowRightFade(!atRightEdge);
+
+          // If the scrollable area is already at the end, hide the right fade initially
+          if (scrollWidth <= clientWidth) {
+            setShowRightFade(false); // If the content fits in the viewport, no need for a fade
+          }
+        }
+
+        // Cleanup event listener on component unmount
+        return () => {
+          if (refCurrent) {
+            refCurrent.removeEventListener("scroll", handleScroll);
+          }
+        };
+      }, [handleScroll]);
+
+      return (
+        <Box as="nav" aria-label="Sub-navigation menu" __css={styles.base}>
+          <Box __css={styles.container}>
+            <Flex
+              alignItems="center"
+              className={className}
+              gap="1rem"
+              justify="space-between"
+            >
+              <List
+                type="ul"
+                m="0"
+                sx={{
+                  ...styles.listCleanup,
+                  ...styles.scrollableList,
+                  ...styles.primaryActions,
+                }}
+                ref={scrollableRef}
+                inline
+                noStyling
+              >
+                {primaryActions}
+                {showRightFade && (
+                  // Explicitly cast styles.fadeEffect to CSSProperties
+                  <div style={styles.fadeEffect as CSSProperties} />
```

Comment:
Ah, @bigfishdesign13 I think this is where the list warning was coming from. This is a div and should be wrapped in an li.

Reply:
@gonuguntla Wrap the div in an `li`. Also, you can convert this to `<Box sx={styles.fadeEffect} />` to remove the type casting for `CSSProperties`.